source/material_model/rheology/diffusion_dislocation.cc

```diff
@@ -88,99 +89,90 @@ namespace aspect
-                        dislocation_strain_rate = diffusion_viscosity / (diffusion_viscosity + dislocation_viscosity) * edot_ii;
-
-                        ++stress_iteration;
-                        AssertThrow(stress_iteration < stress_max_iteration_number,
-                                    ExcMessage("No convergence has been reached in the loop that determines "
-                                               "the ratio of diffusion/dislocation viscosity. Aborting! "
-                                               "Residual is " + Utilities::to_string(log_strain_rate_residual) +
-                                               " after " + Utilities::to_string(stress_iteration) + " iterations. "
-                                               "You can increase the number of iterations by adapting the "
-                                               "parameter 'Maximum strain rate ratio iterations'."));
-
-                        log_strain_rate_residual = std::abs(std::log(diffusion_strain_rate/old_diffusion_strain_rate));
-                        stress_ii = 2.0 * edot_ii * 1./(1./diffusion_viscosity + 1./dislocation_viscosity);
-                      }
-                    while (log_strain_rate_residual > log_strain_rate_residual_threshold);
-
-                    break;
-                  }
-              }
-
-            // The effective viscosity, with minimum and maximum bounds
+            Vector<double> log_stress_ii(1);
+            log_stress_ii[0] = std::log(stress_ii);
+
+            typename SUNDIALS::KINSOL<Vector<double>>::AdditionalData additional_data;
+            additional_data.function_tolerance = log_strain_rate_residual_threshold;
+
+            SUNDIALS::KINSOL<Vector<double>> nonlinear_solver(additional_data);
+
+            nonlinear_solver.reinit_vector = [&](Vector<double> &x)
+            {
+              x.reinit(1);
+            };
+
+            nonlinear_solver.residual =
+              [&](const Vector<double> &evaluation_point,
+                  Vector<double>       &residual)
+            {
+              compute_log_strain_rate_residual(evaluation_point, residual, pressure, temperature, diffusion_creep_parameters, dislocation_creep_parameters, log_edot_ii);
+            };
+
+            nonlinear_solver.setup_jacobian =
+              [&](const Vector<double> &current_u,
+                  const Vector<double> & /*current_f*/)
+            {
+              compute_log_strain_rate_deriv(current_u, pressure, temperature, diffusion_creep_parameters, dislocation_creep_parameters, log_strain_rate_deriv);
+            };
+
+            nonlinear_solver.solve_with_jacobian = [&](const Vector<double> &rhs,
+                                                       Vector<double> &solution,
+                                                       const double /*tolerance*/)
+            {
+              solution(0) -= rhs(0)/log_strain_rate_deriv;
```

Comment:
Should that not be 
```suggestion
              solution(0) = rhs(0)/log_strain_rate_deriv;
```
?